common/persistence/cassandraPersistence.go

```diff
@@ -2310,10 +2310,10 @@ func (d *cassandraPersistence) updateTimerInfos(batch *gocql.Batch, timerInfos [
 	}
 }
 
-func (d *cassandraPersistence) updateChildExecutionInfos(batch *gocql.Batch, childExecutionInfos []*ChildExecutionInfo,
+func (d *cassandraPersistence) updateChildExecutionInfos(batch *gocql.Batch, childExecutionInfos map[*ChildExecutionInfo]struct{},
```

Comment:
looks like you are updating a bunch of parameters which used to be slice to a map.  Is there a good reason, it's not obvious from looking at the change why this is relevant for the bugfix.

Reply:
"bugfix: update mutable state to handle duplicated update activity / user timer for better Cassandra performance"